src/javaharness/java/arcs/webimpl/PortableJsonJsImpl.java

```diff
@@ -10,10 +10,56 @@
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 import jsinterop.base.Any;
+import jsinterop.base.Js;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsFunction;
 
 public class PortableJsonJsImpl implements PortableJson {
     private Any jsonObj;
 
+    @JsMethod(namespace="<window>", name="eval")
+    private native static Object eval(String js);
+
+    @JsFunction
+    interface Getter {
+      Object get(Object obj, String key);
+    }
+
+    @JsFunction
+    interface HasChecker {
+        boolean has(Object obj, String key);
+    }
+
+    @JsFunction
+    interface Setter {
+      Object set(Object obj, String key, Object value);
+    }
+
+    @JsFunction
+    interface Keys {
+        String[] get(Object obj);
+    }
+
+    private <T> T getValue(String key) {
+      Getter getter = (Getter) eval("(obj, key) => { return obj[key]; }");
+      return Js.uncheckedCast(getter.get(jsonObj, key));
+    }
+
+    private boolean hasStringKey(String key) {
+      HasChecker hasChecker = (HasChecker) eval("(obj, key) => { return key in obj; }");
+      return hasChecker.has(jsonObj, key);
+    }
+
+    private <T> T setValue(String key, T value) {
+        Setter setter = (Setter) eval("(obj, key, value) => { obj[key] = value; };");
```

Comment:
This doesn't really return a value. 

Also, I think this method may fail for numbers/booleans as they will be autoboxed into java.lang.Integer/Boolean.

You probably want set(Object obj, String key, Any value)

And when you call it, call set(obj, key, Js.asAny(value))

This will avoid any boxing that occurs from casting from a primitive number type to a method receiving a reference type(e.g. Object)

Reply:
removed return value.
i checked the original and the updated code with:
```
json.put("fooStr", "my-string");
json.put("fooInt", 888);
json.put("fooNum", 9.99);
json.put("fooBool", true);
String fooStr = json.getString("fooStr");
int fooInt = json.getInt("fooInt");
double fooNum = json.getNumber("fooNum");
boolean fooBool = json.getBool("fooBool");
```
and it worked just the same, but I'll stick with your proposal to be safe. thanks!